src/topp/PeptideIndexer.cpp

```diff
@@ -512,14 +173,27 @@ class TOPPPeptideIndexer :
     //-------------------------------------------------------------
     String in = getStringOption_("in");
     String out = getStringOption_("out");
-    bool write_protein_sequence = getFlag_("write_protein_sequence");
-    bool write_protein_description = getFlag_("write_protein_description");
-    bool keep_unreferenced_proteins = getFlag_("keep_unreferenced_proteins");
-    bool allow_unmatched = getFlag_("allow_unmatched");
-    bool il_equivalent = getFlag_("IL_equivalent");
 
-    String decoy_string = getStringOption_("decoy_string");
-    bool prefix = getFlag_("prefix");
+
+    PeptideIndexing indexer;
+
+    Param param = getParam_().copy("", true);
+
+    // These parameters are not used by the PeptideIndexing class
+    param.remove("in");
```

Comment:
agree: update (const Param &old_version, const bool report_new_params=false, const bool only_update_old=false, Logger::LogStream &stream=LOG_WARN)
so I guess update in this case also copies over new ones (as only_update=false is false by default)